### PR TITLE
Issue 18730

### PR DESCRIPTION
**Pull request by erickgonzalez**
This PR is for migrate the template request to the DB from hibernate to dotconnect, now instead of calling the InodeFactory that uses hibernate it calls the TemplateAPI.
Fixes for the import/export util were made to support the removed mapping of hibernate.

Most of the code already had tests but since some  methods were added, Some tests were created.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java [hunk 1]
```diff
@@ -455,23 +440,23 @@ public boolean delete(Template template, User user, boolean respectFrontendRoles
 		}
 	}
 
-	public Template findWorkingTemplate(String id, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
-		VersionInfo info = APILocator.getVersionableAPI().getVersionInfo(id);
+	public Template findWorkingTemplate(final String id, final User user, final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java [hunk 2]
```diff
@@ -220,7 +219,7 @@ Template copy(Template sourceTemplate, Host destinationHost, boolean forceOverwr
 	 * @throws Exception
 	 */
 	@Deprecated
-	public boolean delete(Template template, User user, boolean respectFrontendRoles) throws DotSecurityException, Exception;
+	boolean delete(Template template, User user, boolean respectFrontendRoles) throws DotSecurityException, Exception;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java [hunk 3]
```diff
@@ -482,17 +467,14 @@ public Template find(String inode, User user,  boolean respectFrontEndRoles) thr
 
 	}
 
-
-
-
-	public Template findLiveTemplate(String id, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+	public Template findLiveTemplate(final String id, final User user, final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package com.dotmarketing.portlets.templates.business;
+
+import com.dotmarketing.beans.Inode.Type;
+
+public class TemplateSQL {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java [hunk 7]
```diff
@@ -1,10 +1,14 @@
 package com.dotmarketing.portlets.templates.business;
 
 import com.dotcms.IntegrationTestBase;
+import com.dotcms.content.elasticsearch.business.ESContentFactoryImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESContentFactoryImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java [hunk 8]
```diff
@@ -1,10 +1,14 @@
...
+import com.dotcms.contenttype.model.type.BaseContentType;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.type.BaseContentType'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java:222]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:470]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.model.type.BaseContentType'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/DotTemplateTool.java [hunk 9]
```diff
@@ -132,15 +134,16 @@ public static void removeFromLayoutCache(final String templateInode){
...
-        final Template template = APILocator.getTemplateAPI().findWorkingTemplate(ident.getId(), user, false);
+    private static DrawedBody getDrawedBody(String themeInodeOrId, User user) throws DotDataException, DotSecurityException {
+        final Template template = FactoryLocator.getTemplateFactory().find(themeInodeOrId);//If null themeInodeOrId is a Id
+        final String identifier = template!=null ? template.getIdentifier() : themeInodeOrId;
+        final Template workingTemplate = APILocator.getTemplateAPI().findWorkingTemplate(identifier, user, false);
 
-        if (!template.isDrawed()){
-            throw new RuntimeException("Template with inode: " + themeInode + " is not drawed");
+        if (!workingTemplate.isDrawed()){
+            throw new RuntimeException("Template with inode: " + themeInodeOrId + " is not drawed");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/DotTemplateTool.java:143]: Codacy found an issue: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/templates/business/TemplateAPITest.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESContentFactoryImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java:443]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:5]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/DotTemplateTool.java [hunk 10]
```diff
@@ -21,6 +22,7 @@
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.model.User;
 
+import io.vavr.control.Try;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/DotTemplateTool.java:25]: Codacy found an issue: [Avoid unused imports such as 'io.vavr.control.Try'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380036)

File: dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    public static final String FIND_WORKING_TEMPLATE_BY_HOST_INODE_AND_TITLE =
+            "select template.*, template_1_.* from " + Type.TEMPLATE.getTableName() + " template, inode template_1_, " +
+                    "identifier template_identifier, " + Type.TEMPLATE.getVersionTableName() + " vi where " +
+                    "template_identifier.host_inode = ? and template_identifier.id = template.identifier and " +
+                    "vi.identifier=template_identifier.id and template.title = ? and " +
+                    "template.inode = template_1_.inode and " +
+                    "template.inode=vi.working_inode ";
+
+//    public static final String SELECT_ALL_FIELDS = "select inode, show_on_menu, title, mod_date, mod_user, " +
+//            "sort_order, friendly_name, body, header, footer, image, identifier, drawed, drawed_body, " +
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateSQL.java:25]: remove this

**Review by nollymar — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**